lib/IRGen/GenDecl.cpp

```diff
@@ -6258,6 +6258,7 @@ IRGenModule::getAddrOfContinuationPrototype(CanSILFunctionType fnType) {
   llvm::Function *&entry = GlobalFuncs[entity];
   if (entry) return entry;
 
```

Comment:
I think that adding a GenericContextScope is generally considered a code smell, as it sets some global state to workaround issues with type substitution.

@slavapestov Am I remembering this right?

Reply:
I am taking this comment "back".

The other Signature expansion APIs add a GenericContextScope as well (e.g in `Signature::getUncached`)